### ISSUE TITLE
Existing SIP account setup failing to login

### ISSUE THREAD

**Issue by matthewepler** [BelledonneCommunications/linphone-android#1331]
**Describe the bug**
When attempting to add an existing SIP account, pressing the "Login" button produces no results. No error, no update to UI, no feedback. Just stays on the form page and does nothing. 

I assume this is an error to do with my input values, which I have included below.

**To Reproduce**
1. Go to 'Assistant', 
2. Click on 'Use SIP Account'
3. Fill out the form
4. Tap "Login"

**Expected behavior**
Either:
a) Login attempt is made and feedback on the result is show
or
b) Error is shown with details on how to fix it

**Please complete the following information**
 - Device: Google Pixel 4a
 - OS: GrapheneOS (latest)
 - Version of the App 4.3.1
 - Version of the SDK 4.4.2

**Additional context**
Form input:
Username: betafish~40dismail.de
Password: (three words separated by spaces, also tried without spaces)
Domain: jmp.bwapp.bwsip.io

**Comment by matthewepler** [comment]
After some troubleshooting, the issue seems to be the `~` character in the userID.

**Comment by Viish** [comment]
Glad you found the issue by yourself.
By the way the next version of the app (cf feature/kotlin_rewrite) branch has better error handling for such scenario.

**Comment by matthewepler** [comment]
Thank you. FYI, the service jmp.chat seems to create usernames that will always have a `~` symbol in them. It would be great if we could use Linphone with these usernames. Is there a reason this character cannot be used? Can you allow it as part of a string?

**Comment by rOOb85** [comment]
I had the same exact issue with the same provider(jmp.chat).

I found a workaround:
Setup a sip account like normal but if the username contains ~ don't add it. Hit login and linphone moves on to the next screen.
Open the sidedrawer and click the account name at the very top. This brings up the account settings screen.
Change the username to the correct username that includes ~
I also had to change Transport to TCP.

Linphone that correctly connects and works as expected. Hope this helps!
